From 9e9de34603ab7256b937fb06b7f5d9a073c3c47f Mon Sep 17 00:00:00 2001
From: Neha Bisht <nbisht@qti.qualcomm.com>
Date: Wed, 11 Sep 2024 10:28:22 +0530
Subject: [PATCH] QSDK: wifi: ath12k: Update encap/decap to ethernet for STA

In monitor + station scenario, DS iperf traffic doesn't go through.
This is due to encap type for both monitor and STA is set to
native wifi.
In DS, we update the vif offload type from mac80211 for AP
mode only. Add support to update the offload type and hence
encap to ethernet for STA as well.

Patch-dependency: 951-c-QSDK-wifi-ath12k-support-DS-for-SLO-and-MLO-AP-modes.patch

Signed-off-by: Neha Bisht <nbisht@qti.qualcomm.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 335ba59..46c4c3c 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -11783,7 +11783,7 @@ static void ath12k_mac_update_vif_offload(struct ath12k *ar,
 #ifdef CPTCFG_ATH12K_PPE_DS_SUPPORT
 	/* TODO: DS: revisit this for DS support in WDS mode */
 	if (test_bit(ATH12K_FLAG_PPE_DS_ENABLED, &ab->dev_flags) &&
-	    vif->type == NL80211_IFTYPE_AP)
+	    (vif->type == NL80211_IFTYPE_AP || vif->type == NL80211_IFTYPE_STATION))
 		vif->offload_flags |= (IEEE80211_OFFLOAD_ENCAP_ENABLED |
 					IEEE80211_OFFLOAD_DECAP_ENABLED);
 #endif
-- 
2.34.1

